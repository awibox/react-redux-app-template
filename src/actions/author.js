import axios from 'axios';
import {GET_AUTHOR_SUCCESS} from '../actions/types';
import {getError} from "./error";
import {gitHubApiUrl, authorAccount} from '../../configs/config';

const END_POINT = 'users';

const getAuthorSuccess = author => ({
    type: GET_AUTHOR_SUCCESS,
    payload: author
});

export const getAuthor = () => dispatch => {
    axios.get(`${gitHubApiUrl}/${END_POINT}/${authorAccount}`)
        .then(response => {
            const {data} = response;
            dispatch(getAuthorSuccess(data));
        }).catch(err => {
            dispatch(getError(err.response.data))
        });
};
