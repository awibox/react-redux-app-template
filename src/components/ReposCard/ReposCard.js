import React from 'react';
import PropTypes from 'prop-types';
// Components
import { Card } from 'tigerspack';
// Styles
import styles from './ReposCard.scss';

// Using "Stateless Functional Components"
const ReposCard = (props) => {
  const { repo, style } = props;
  return (
    <div className={styles.repo}>
      {repo.id > 0
        && <Card theme={'light'}>
          <a href={repo.html_url} rel="noopener noreferrer" target="_blank" className={styles.name}>{repo.name}</a>
          <div className={styles.githubIcon}><i className="fa fa-github" aria-hidden="true"></i></div>
          <div className={styles.desc}>{repo.description}</div>
          <div className={styles.panel}>
            {repo.language
            && <div className={styles.property}><i className="fa fa-circle" style={style}></i> {repo.language}</div>}
            <div className={styles.size}><i className="fa fa-file-archive-o"></i> {(repo.size / 1024).toFixed(2)} mb
            </div>
            <div className={styles.stars}><i className="fa fa-star" aria-hidden="true"></i> {repo.stargazers_count}
            </div>
          </div>
        </Card>
      }
    </div>
  );
};

ReposCard.propTypes = {
  repo: PropTypes.shape({
    html_url: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    description: PropTypes.string,
    language: PropTypes.string,
    id: PropTypes.number.isRequired,
    size: PropTypes.number,
    stargazers_count: PropTypes.number,
  }),
  style: PropTypes.object,
};

export default ReposCard;
