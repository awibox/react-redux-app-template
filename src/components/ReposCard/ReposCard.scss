@import "../../styles/variables";

.repo {
  padding-left: $defaultIndent;
  position: relative;
  width: 20%;
  box-sizing: border-box;
}

.name {
  color: #4078c0;
  font-size: 16px;
}

.desc {
  font-size: 14px;
  margin-bottom: 16px;
  margin-top: 8px;
  color: #767676;
  font-weight: 100;
  min-height: 18px;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.panel {
  display: flex;
  flex-flow: row;
}

.property {
  color: #767676;
  font-weight: 100;
  margin-right: 50px;
}

.fa {
  margin-right: 5px;
}

.stars {
  @extend .property;
  flex: 1;
  text-align: right;
  margin-right: 0;
  .fa {
    margin-right: 0;
  }
}

.size {
  @extend .property;
}

.githubIcon {
  font-size: 30px;
  position: absolute;
  top: 10px;
  right: 10px;
}
@media(max-width: $widthScreenDisplayLarge) {
  .repo {
    width: 25%;
  }
}
@media(max-width: $widthScreenDisplayDefault) {
  .repo {
    width: 33.33%;
  }
}
@media(max-width: $widthScreenLaptopSmall) {
  .repo {
    width: 50%;
  }
}
@media(max-width: $widthScreenTabletPortrait) {
  .repo {
    width: 100%;
  }
}

