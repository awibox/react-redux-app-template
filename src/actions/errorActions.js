import { GET_ERRORS } from 'actions/types';

export const getError = (error) => ({
  type: GET_ERRORS,
  payload: error,
});

export default getError;
