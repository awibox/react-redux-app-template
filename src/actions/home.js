import axios from 'axios';
import {GET_HOME_SUCCESS} from 'actions/types';
import {gitHubApiUrl, authorAccount, repoName} from 'config';
import {getError} from './error';

const END_POINT = 'repos';


export const getHomeSuccess = (home) => ({
  type: GET_HOME_SUCCESS,
  home,
});

export const getHome = () => async (dispatch) => {
  try {
    const {data} = await axios.get(`${gitHubApiUrl}/${END_POINT}/${authorAccount}/${repoName}`);
    dispatch(getHomeSuccess(data));
  } catch (err) {
    dispatch(getError(err.response.data));
  }
};
