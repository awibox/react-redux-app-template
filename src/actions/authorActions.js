import axios from 'axios';
import { trackPromise } from 'react-promise-tracker';
import { GET_AUTHOR_SUCCESS } from 'actions/types';
import { gitHubApiUrl, authorAccount } from 'config';
import { getError } from 'actions/errorActions';


const END_POINT = 'users';

export const getAuthorSuccess = (author) => ({
  type: GET_AUTHOR_SUCCESS,
  payload: author,
});

export const getAuthorAction = () => async (dispatch) => {
  try {
    const { data } = await trackPromise(axios.get(`${gitHubApiUrl}/${END_POINT}/${authorAccount}`));
    dispatch(getAuthorSuccess(data));
  } catch (err) {
    dispatch(getError(err.response.data));
  }
};
