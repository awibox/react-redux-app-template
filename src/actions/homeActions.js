import axios from 'axios';
import { trackPromise } from 'react-promise-tracker';
import { GET_HOME_SUCCESS } from 'actions/types';
import { gitHubApiUrl, authorAccount, repoName } from 'config';
import { getError } from './errorActions';

const END_POINT = 'repos';

export const getHomeSuccess = (home) => ({
  type: GET_HOME_SUCCESS,
  payload: home,
});

export const getHomeAction = () => async (dispatch) => {
  try {
    const { data } = await trackPromise(axios.get(`${gitHubApiUrl}/${END_POINT}/${authorAccount}/${repoName}`));
    dispatch(getHomeSuccess(data));
  } catch (err) {
    dispatch(getError(err.response.data));
  }
};
