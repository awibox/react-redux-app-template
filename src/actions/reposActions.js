import axios from 'axios';
import { gitHubApiUrl } from 'config';
import { getError } from 'actions/errorActions';
import { setLoadStatus } from 'actions/loadActions';
import { GET_REPOS_SUCCESS } from './types';

const END_POINT = 'users';

export const getReposSuccess = (repos) => ({
  type: GET_REPOS_SUCCESS,
  payload: repos,
});

export const getReposAction = (user) => async (dispatch) => {
  dispatch(setLoadStatus(true));
  try {
    const { data } = await axios.get(`${gitHubApiUrl}/${END_POINT}/${user}/repos`);
    dispatch(getReposSuccess(data));
  } catch (err) {
    dispatch(getError(err.response.data));
  }
  dispatch(setLoadStatus(false));
};
