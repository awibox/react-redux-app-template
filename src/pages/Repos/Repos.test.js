import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import ReposCard from 'components/ReposCard/ReposCard';
import { Alert } from 'tigerspack';
import { List } from 'immutable';
import { getReposSuccess } from 'actions/reposActions';
import ReposContainer from './Repos';

describe('Repos', () => {
  const mockData = {
    html_url: 'http://localhost:8888/item',
    name: 'First Repo',
    description: 'Repository',
    language: 'javascript',
    id: 1,
    size: 10,
    stargazers_count: 10,
  };
  const initialState = {
    reposState: {
      repos: List([
        mockData,
      ]),
    },
    errors: {
      message: 'Fake error',
    },
    load: {
      isLoading: false,
    },
    match: {
      params: {
        user: 'test',
      },
    },
    getReposAction: () => {},
  };
  const mockStore = configureStore([thunk]);
  let store;
  let container;

  beforeEach(() => {
    store = mockStore(initialState);
    container = mount(<Provider store={store}>
      <Router>
        <ReposContainer getReposAction={initialState.getReposAction} match={initialState.match}/>
      </Router>
    </Provider>);
  });
  it('should render coorectly', () => {
    expect(container).toMatchSnapshot();
  });
  it('should render the container', () => {
    expect(container.length).toEqual(1);
  });
  it('should matches with initialState in the ReposCard', () => {
    expect(container.find(ReposCard).at(0).prop('repo')).toEqual(mockData);
  });
  it('should display Alert', () => {
    expect(container.find(Alert)).toExist();
  });
  it('should check prop Alert', () => {
    expect(container.find(Alert).prop('children')).toEqual(initialState.errors.message);
  });
  it('should action on dispatching', () => {
    store.dispatch(getReposSuccess(mockData));
    const action = store.getActions();
    expect(action[0].type).toBe('GET_REPOS_SUCCESS');
    expect(action[0].payload).toEqual(mockData);
    container.unmount();
  });
});
