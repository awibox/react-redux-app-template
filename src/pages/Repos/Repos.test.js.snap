// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repos should render coorectly 1`] = `
<div
  className="repos"
>
  <div
    className="css-ciu1tw"
    style={
      Object {
        "display": "block",
      }
    }
  >
    Fake error
  </div>
  <div
    className="title"
  >
    Select the user
  </div>
  <div
    className="css-wk4os9"
  >
    <div
      className="css-old1by"
    >
      <div
        className="users"
      >
        <a
          aria-current={null}
          className="link"
          href="/repos/awibox"
          onClick={[Function]}
        >
          <div
            className="user"
          >
            <i
              className="fa fa-user"
            />
            awibox
          </div>
        </a>
        <a
          aria-current={null}
          className="link"
          href="/repos/angular"
          onClick={[Function]}
        >
          <div
            className="user"
          >
            <i
              className="fa fa-user"
            />
            angular
          </div>
        </a>
        <a
          aria-current={null}
          className="link"
          href="/repos/facebook"
          onClick={[Function]}
        >
          <div
            className="user"
          >
            <i
              className="fa fa-user"
            />
            facebook
          </div>
        </a>
        <a
          aria-current={null}
          className="link"
          href="/repos/reduxjs"
          onClick={[Function]}
        >
          <div
            className="user"
          >
            <i
              className="fa fa-user"
            />
            reduxjs
          </div>
        </a>
        <a
          aria-current={null}
          className="link"
          href="/repos/webpack"
          onClick={[Function]}
        >
          <div
            className="user"
          >
            <i
              className="fa fa-user"
            />
            webpack
          </div>
        </a>
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "height": "20px",
        "width": "100%",
      }
    }
  />
  <div
    className="title"
  >
    Repositories of 
    test
  </div>
  <div
    className="reposList"
  >
    <div
      className="repo"
    >
      <div
        className="css-wk4os9"
      >
        <div
          className="css-1v1qs4"
        >
          <a
            className="name"
            href="http://localhost:8888/item"
            rel="noopener noreferrer"
            target="_blank"
          >
            First Repo
          </a>
          <div
            className="githubIcon"
          >
            <i
              aria-hidden="true"
              className="fa fa-github"
            />
          </div>
          <div
            className="desc"
          >
            Repository
          </div>
          <div
            className="panel"
          >
            <div
              className="property"
            >
              <i
                className="fa fa-circle"
                style={Object {}}
              />
               
              javascript
            </div>
            <div
              className="size"
            >
              <i
                className="fa fa-file-archive-o"
              />
               
              0.01
               mb
            </div>
            <div
              className="stars"
            >
              <i
                aria-hidden="true"
                className="fa fa-star"
              />
               
              10
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "backfaceVisibility": "hidden",
        "background": "rgba(255,255,255,.5)",
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "zIndex": 10,
      }
    }
  >
    <div
      style={
        Object {
          "height": 100,
          "left": "50%",
          "position": "absolute",
          "top": "50%",
          "transform": "translate(-50%, -50%)",
          "width": 100,
        }
      }
    >
      <svg
        className="lds-spinner"
        height="100%"
        preserveAspectRatio="xMidYMid"
        viewBox="0 0 100 100"
        width="100%"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          key="rotate(0 50 50)"
          transform="rotate(0 50 50)"
        >
          <rect
            fill="#3F88C5"
            height="12"
            rx="9.4"
            ry="4.8"
            width="6"
            x="47"
            y="24"
          >
            <animate
              attributeName="opacity"
              begin="-0.9166666666666666s"
              dur="1s"
              keyTimes="0;1"
              repeatCount="indefinite"
              values="1;0"
            />
          </rect>
        </g>
        <g
          key="rotate(30 50 50)"
          transform="rotate(30 50 50)"
        >
          <rect
            fill="#3F88C5"
            height="12"
            rx="9.4"
            ry="4.8"
            width="6"
            x="47"
            y="24"
          >
            <animate
              attributeName="opacity"
              begin="-0.8333333333333334s"
              dur="1s"
              keyTimes="0;1"
              repeatCount="indefinite"
              values="1;0"
            />
          </rect>
        </g>
        <g
          key="rotate(60 50 50)"
          transform="rotate(60 50 50)"
        >
          <rect
            fill="#3F88C5"
            height="12"
            rx="9.4"
            ry="4.8"
            width="6"
            x="47"
            y="24"
          >
            <animate
              attributeName="opacity"
              begin="-0.75s"
              dur="1s"
              keyTimes="0;1"
              repeatCount="indefinite"
              values="1;0"
            />
          </rect>
        </g>
        <g
          key="rotate(90 50 50)"
          transform="rotate(90 50 50)"
        >
          <rect
            fill="#3F88C5"
            height="12"
            rx="9.4"
            ry="4.8"
            width="6"
            x="47"
            y="24"
          >
            <animate
              attributeName="opacity"
              begin="-0.6666666666666666s"
              dur="1s"
              keyTimes="0;1"
              repeatCount="indefinite"
              values="1;0"
            />
          </rect>
        </g>
        <g
          key="rotate(120 50 50)"
          transform="rotate(120 50 50)"
        >
          <rect
            fill="#3F88C5"
            height="12"
            rx="9.4"
            ry="4.8"
            width="6"
            x="47"
            y="24"
          >
            <animate
              attributeName="opacity"
              begin="-0.5833333333333334s"
              dur="1s"
              keyTimes="0;1"
              repeatCount="indefinite"
              values="1;0"
            />
          </rect>
        </g>
        <g
          key="rotate(150 50 50)"
          transform="rotate(150 50 50)"
        >
          <rect
            fill="#3F88C5"
            height="12"
            rx="9.4"
            ry="4.8"
            width="6"
            x="47"
            y="24"
          >
            <animate
              attributeName="opacity"
              begin="-0.5s"
              dur="1s"
              keyTimes="0;1"
              repeatCount="indefinite"
              values="1;0"
            />
          </rect>
        </g>
        <g
          key="rotate(180 50 50)"
          transform="rotate(180 50 50)"
        >
          <rect
            fill="#3F88C5"
            height="12"
            rx="9.4"
            ry="4.8"
            width="6"
            x="47"
            y="24"
          >
            <animate
              attributeName="opacity"
              begin="-0.4166666666666667s"
              dur="1s"
              keyTimes="0;1"
              repeatCount="indefinite"
              values="1;0"
            />
          </rect>
        </g>
        <g
          key="rotate(210 50 50)"
          transform="rotate(210 50 50)"
        >
          <rect
            fill="#3F88C5"
            height="12"
            rx="9.4"
            ry="4.8"
            width="6"
            x="47"
            y="24"
          >
            <animate
              attributeName="opacity"
              begin="-0.3333333333333333s"
              dur="1s"
              keyTimes="0;1"
              repeatCount="indefinite"
              values="1;0"
            />
          </rect>
        </g>
        <g
          key="rotate(240 50 50)"
          transform="rotate(240 50 50)"
        >
          <rect
            fill="#3F88C5"
            height="12"
            rx="9.4"
            ry="4.8"
            width="6"
            x="47"
            y="24"
          >
            <animate
              attributeName="opacity"
              begin="-0.25s"
              dur="1s"
              keyTimes="0;1"
              repeatCount="indefinite"
              values="1;0"
            />
          </rect>
        </g>
        <g
          key="rotate(270 50 50)"
          transform="rotate(270 50 50)"
        >
          <rect
            fill="#3F88C5"
            height="12"
            rx="9.4"
            ry="4.8"
            width="6"
            x="47"
            y="24"
          >
            <animate
              attributeName="opacity"
              begin="-0.16666666666666666s"
              dur="1s"
              keyTimes="0;1"
              repeatCount="indefinite"
              values="1;0"
            />
          </rect>
        </g>
        <g
          key="rotate(300 50 50)"
          transform="rotate(300 50 50)"
        >
          <rect
            fill="#3F88C5"
            height="12"
            rx="9.4"
            ry="4.8"
            width="6"
            x="47"
            y="24"
          >
            <animate
              attributeName="opacity"
              begin="-0.08333333333333333s"
              dur="1s"
              keyTimes="0;1"
              repeatCount="indefinite"
              values="1;0"
            />
          </rect>
        </g>
        <g
          key="rotate(330 50 50)"
          transform="rotate(330 50 50)"
        >
          <rect
            fill="#3F88C5"
            height="12"
            rx="9.4"
            ry="4.8"
            width="6"
            x="47"
            y="24"
          >
            <animate
              attributeName="opacity"
              begin="0s"
              dur="1s"
              keyTimes="0;1"
              repeatCount="indefinite"
              values="1;0"
            />
          </rect>
        </g>
      </svg>
    </div>
  </div>
</div>
`;
